### ISSUE TITLE
CVRPTW Vehicle start/end time

### ISSUE THREAD

**Issue by Migaelh** [google/or-tools#1052]
Hi guys,

I am using Capacitated Vehicle Routing Problem with Time Windows model (C#). I use drivers (vehicles) that works in shifts (start time and end time), so I need to set the time windows **for the vehicle**. 

Is there a way to do this?

I have tried this, 

```c#
for (int vehicle_id = 0; vehicle_id < num_vehicles; ++vehicle_id)
{ 
            long start = vehicles[vehicle_id].shift_start;
            long end = vehicles[vehicle_id].shift_end;
            long index = routing.Start(vehicle_id);
            time_dimension.CumulVar(index).SetRange(start, end);
            routing.AddToAssignment(time_dimension.SlackVar(index));
}
```

but the solution does not start with the vehicle's "shift_start" time, but with the first stop's time window (start time). Also, the solution ends with the vehicle's end location but outside the vehicle's time window.  

Thanks!

**Comment by Mizux** [comment]
You must set a Range on the end node.
```cs
  long index = routing.End(vehicle_id);
  time_dimension.CumulVar(index).SetRange(start, end);
```

**Comment by murr999888** [comment]
Establishing a start and end time for a vehicle is not a resolved issue.

**CVRPTW**
**C#**
I have 350-400 orders by day and 21 cars. Setting a time window for the end point of the route works on small examples, but does not work on large numbers. 

>  long index = routing.End(vehicle_id);
  time_dimension.CumulVar(index).SetRange(start, end);

Some vehicles may only work part of the day. If I set the arrival time window for one or two cars, then after a long wait I get the expected result: the time window defined for the car is filled with orders, part of the orders are dropped, the rest of the orders are redistributed between the other vehicles. But if I set the time window for a larger number (> 2-3) of cars, then after a longlongLONG wait **all** orders are discarded, despite the large penalty.

At the same time, determining the time window for the starting point of the route is fast and expected.
